### PR TITLE
Adopt `NodeSnapshot.func_name` from Snapshot

### PR DESCRIPTION
**Pull request by jitu5**
**Sub-task:** #2715 · **Parent:** #2265

## Summary

Uses `NodeSnapshot.func_name` to reproduce the existing Kedro-Viz task node IDs from inspection snapshots. This keeps the snapshot adapter compatible with the legacy graph and run-status IDs.

## Changes

- Pins Kedro to the commit that introduces `NodeSnapshot.func_name`.
- Reconstructs the legacy node string before hashing it.
- Handles automatic names, explicit names, namespaces and partial functions.
- Uses `func_name` when determining task display names.
- Adds ID parity tests against the captured legacy baseline.
- Updates the node-ID baseline report.

Dataset IDs remain unchanged.

## Testing

Tests cover:

- ID parity across every demo pipeline.
- Automatic and explicit node names.
- Namespaced and partial-function nodes.
- Nodes with empty inputs or outputs.
- Display-name behaviour.

## Release requirement

The temporary Kedro Git pin must be replaced with the released Kedro version containing `NodeSnapshot.func_name` before publishing Kedro-Viz.

### CODE REVIEW

File: package/requirements.txt [hunk 1]
```diff
@@ -3,7 +3,8 @@ click-default-group
 fastapi>=0.100.0,<0.200.0
 fsspec>=2021.4
 ipython>=7.0.0, <9.0
-kedro>=1.4.0
+# Replace with the released Kedro version containing NodeSnapshot.func_name before release.
```
Comment (ravi-kumar-pilla) [inline package/requirements.txt]:
```suggestion
# [TODO] Replace with the released Kedro version containing NodeSnapshot.func_name before release.
```

File: package/kedro_viz/integrations/kedro/node_ids.py [hunk 2]
```diff
@@ -2,11 +2,22 @@
...
+import re
 from collections.abc import Sequence
 
 from kedro_viz.utils import _hash, _hash_input_output
 
+_AUTO_NAME_RE = re.compile(r"^(?P<base>.+)__[0-9a-f]{8}$")
+
+
+def _is_auto_node_name(local_name: str, func_name: str) -> bool:
+    """Whether a local snapshot name has Kedro's generated node-name form."""
```
Comment (ravi-kumar-pilla) [inline package/kedro_viz/integrations/kedro/node_ids.py]: This is a bit unclear for me. Can you add an example in the doc string for readability or may be rename the function.

> Whether a local snapshot name has Kedro's generated node-name form.

The doc string uses local snapshot name - what does that mean ?
Kedro's generated node-name form - can you please explain a bit what you meant ?

File: package/kedro_viz/integrations/kedro/node_ids.py [hunk 3]
```diff
@@ -2,11 +2,22 @@
...
+    match = _AUTO_NAME_RE.fullmatch(local_name)
+    if match is None:
+        return False
+    base = match["base"]
+    return base == func_name or (base.startswith("partial(") and base.endswith(")"))
```
Comment (ravi-kumar-pilla) [inline package/kedro_viz/integrations/kedro/node_ids.py]: **Suggestion:** The partial branch treats any `partial(...)__<8hex>` name as auto-generated without checking `func_name`:

```python
return base == func_name or (base.startswith("partial(") and base.endswith(")"))
```

This matches Kedro's `_set_unique_name` for partials, but would misclassify a user who explicitly sets a name like `partial(foo)__deadbeef`. Unlikely in practice — a one-line comment explaining why partials are special-cased would make the intent clearer.

File: package/kedro_viz/integrations/kedro/node_ids.py [hunk 4]
```diff
@@ -17,21 +28,26 @@ def _create_dataset_node_id(dataset_name: str) -> str:
...
+    namespace: str | None,
+    inputs: Sequence[str],
+    outputs: Sequence[str],
 ) -> str:
-    """Return the Viz graph ID for a task node.
-
-    Hashes only identity-defining fields: the namespaced node name plus its inputs and outputs.
-    The fields are JSON-serialized so the encoding is unambiguous (e.g. a single input
-    ``"a,b"`` never collides with two inputs ``"a"``, ``"b"``).
+    """Reconstruct and hash ``Node.__str__`` for the legacy Viz task ID.
```
Comment (ravi-kumar-pilla) [inline package/kedro_viz/integrations/kedro/node_ids.py:57]: **Suggestion:** Consider noting a reconstruction limitation in this docstring: if a user explicitly names a node using Kedro's auto-name pattern (e.g. `clean_data__a1b2c3d4` with func `clean_data`), `_is_auto_node_name` returns `True` and the prefix is omitted, but live `Node.__str__` includes `{name}: ` because `_name` is set. The snapshot has no explicit-vs-auto flag, so this ambiguity is inherent — worth documenting so future readers don't chase it as a bug.

File: package/kedro_viz/integrations/kedro/inspection/graph_builder.py [hunk 5]
```diff
@@ -287,14 +297,12 @@ def _add_edge(
...
 
-def _display_name(snapshot_name: str, namespace: str | None) -> str:
-    """Derive the UI display name: strip the namespace and any auto-name ``__<hash>`` suffix."""
-    local = snapshot_name
-    prefix = f"{namespace}."
-    if namespace and local.startswith(prefix):
-        local = local[len(prefix) :]
-    auto = _AUTO_NAME_RE.match(local)
-    return auto.group("func") if auto else local
+def _display_name(snapshot_name: str, func_name: str, namespace: str | None) -> str:
```
Comment (ravi-kumar-pilla) [inline package/kedro_viz/integrations/kedro/inspection/graph_builder.py:300]: **Suggestion:** `_display_name` could use Google-style `Args:` / `Returns:` sections for consistency with `_create_task_node_id` and the rest of the inspection adapter.

File: package/tests/test_inspection_adapter/test_ids.py [hunk 6]
```diff
@@ -21,47 +21,87 @@ def task_nodes() -> list[dict]:
...
 
 
-def test_stored_graph_id_matches_current_scheme(task_nodes: list[dict]) -> None:
-    """Test that each stored ``graph_id`` still matches what ``_create_task_node_id`` computes."""
-    for node in task_nodes:
-        assert node["graph_id"] == _task_id(node), (
-            f"stale baseline for {node['snapshot_name']!r}: regenerate "
-            "node_id_report.json via capture_baseline.py"
-        )
+def test_task_node_ids_are_unique(task_nodes: list[dict]) -> None:
```
Comment (ravi-kumar-pilla) [inline package/tests/test_inspection_adapter/test_ids.py:43]: **Suggestion:** `test_task_node_id_changes_with_io` was removed in this PR. The parity tests against the demo baseline cover real pipelines well, but a small focused unit test asserting that changing inputs or outputs changes the hash under the new `Node.__str__` scheme would guard against accidental regressions in `_create_task_node_id`.

**Review by ravi-kumar-pilla — COMMENTED** [review]
## Kedro PR Review
> Generated with `review-kedro-pr` skill.

### Suggestions
- [ ] `node_ids.py:L19` — Comment why partial names bypass the `func_name` equality check
- [ ] `node_ids.py:L37` — Document the explicit-name-that-looks-auto edge case
- [ ] `graph_builder.py:L300` — Add Google-style docstring to `_display_name`
- [ ] `test_ids.py:L43` — Re-add a focused IO-sensitivity unit test

**Comment by ravi-kumar-pilla** [comment]
Hi @jitu5 , Overall it looks good. Added some minor suggestions both after manual review and from skill review. Thank you 👍

File: package/kedro_viz/integrations/kedro/node_ids.py (hunk 2 above)
Comment (jitu5) [reply package/kedro_viz/integrations/kedro/node_ids.py]: Thanks, updated. The docstring now explains that the namespace has already been removed and includes an example of Kedro’s generated name format.

"local snapshot name" meant the node name with the namespace prefix stripped, e.g. `ingestion.clean_data` becomes `clean_data`. I've reworded it to say that plainly.

"Kedro's generated node-name form" is what Kedro creates when a node is defined without `name=. Node._set_unique_name()` produces `<base>__<8 hex chars>`, where base is the function name, or partial(<function>) for partial functions.

File: package/kedro_viz/integrations/kedro/node_ids.py (hunk 3 above)
Comment (jitu5) [reply package/kedro_viz/integrations/kedro/node_ids.py]: done

File: package/kedro_viz/integrations/kedro/inspection/graph_builder.py (hunk 5 above)
Comment (jitu5) [reply package/kedro_viz/integrations/kedro/inspection/graph_builder.py:300]: done

File: package/tests/test_inspection_adapter/test_ids.py (hunk 6 above)
Comment (jitu5) [reply package/tests/test_inspection_adapter/test_ids.py:43]: done.

File: package/kedro_viz/integrations/kedro/node_ids.py (hunk 4 above)
Comment (jitu5) [reply package/kedro_viz/integrations/kedro/node_ids.py:57]: done

**Review by ravi-kumar-pilla — APPROVED** [review]
LGTM...please add a note about the TODO in feat/backend_v2 or the parent ticket which can be addressed before merging (all the TODOs till now). Thank you @jitu5

**Review by SajidAlamQB — APPROVED** [review]
Looks good, thank you @jitu5!
